CMakeLists.txt

```diff
@@ -53,13 +53,23 @@ endif()
 
 # FreeBSD has a reimplementation of alsalib, so don't restrict the ALSA backend strictly to Linux.
 if(UNIX AND NOT APPLE)
-  find_package(ALSA REQUIRED)
-  target_sources(PortTime PRIVATE src/porttime/ptlinux.c)
-  target_sources(PortMidi PRIVATE
-    src/portmidi/linux/pmlinux.c
-    src/portmidi/linux/pmlinuxalsa.c
-    src/portmidi/linux/finddefault.c)
-  target_link_libraries(PortMidi PRIVATE ALSA::ALSA)
+  if(ALSA_SUPPORT)
```

Comment:
We need to specify defaults for these. It looks like this will break the build without either ALSA_SUPPORT or SNDIO_SUPPORT specified. I propose to check if the OS is a BSD and default to SNDIO in that case and use ALSA in other cases.

Should ALSA and SNDIO support be mutually exclusive? I don't know what would happen if you tried running both on BSD. Could you test that?

Reply:
Yes this breaks the Ubuntu build on GitHub Actions. It does not compile the ALSA backend:
```
 [ 20%] Building C object CMakeFiles/PortTime.dir/src/porttime/porttime.c.o
[ 40%] Linking C shared library libporttime.so
[ 40%] Built target PortTime
[ 60%] Building C object CMakeFiles/PortMidi.dir/src/portmidi/common/portmidi.c.o
[ 80%] Building C object CMakeFiles/PortMidi.dir/src/portmidi/common/pmutil.c.o
[100%] Linking C shared library libportmidi.so
[100%] Built target PortMidi
```